ecl/hthor/hthor.cpp

```diff
@@ -958,11 +955,20 @@ void CHThorXmlWriteActivity::setFormat(IFileDescriptor * desc)
     desc->queryProperties().setProp("@format","utf8n");
     desc->queryProperties().setProp("@rowTag",rowTag.str());
     desc->queryProperties().setProp("@kind", (kind==TAKjsonwrite) ? "json" : "xml");
-    if (headerLength)
+    if (kind==TAKjsonwrite)
+    {
         desc->queryProperties().setPropInt("@headerLength", headerLength);
-    if (footerLength)
         desc->queryProperties().setPropInt("@footerLength", footerLength);
+    }
+    else
+    {
+        // If multiple XML files with different header/footer sprayed into a
+        // multi slaves Thor system then every parts can have different length
+        // header/footer. For compatibility reason Hthor does same.
+        desc->queryPart(0)->queryProperties().setPropInt("@headerLength", headerLength);
```

Comment:
Again if for files for which each part has the same header length you write it to the file properties instead of each part, then this can be the same as json.  Because ECL writes files where the header and footer are all the same.

This makes it clear to me why it should be in the File attributes whenever possible.